pkg/assembler/backends/arangodb/path_test.go

```diff
@@ -983,3 +983,1832 @@ func Test_Nodes(t *testing.T) {
+	}
+	tests := []struct {
+		name                     string
+		pkgInput                 *model.PkgInputSpec
+		srcInput                 *model.SourceInputSpec
+		vulnInput                *model.VulnerabilityInputSpec
+		licenseInput             *model.LicenseInputSpec
+		inPkg                    []*model.PkgInputSpec
+		inSrc                    []*model.SourceInputSpec
+		inArt                    []*model.ArtifactInputSpec
+		inVuln                   []*model.VulnerabilityInputSpec
+		inBld                    []*model.BuilderInputSpec
+		inLic                    []*model.LicenseInputSpec
+		queryArtifactID          bool
+		queryEqualArtifactID     bool
+		queryBuilderID           bool
+		queryPkgTypeID           bool
+		queryPkgNamespaceID      bool
+		queryPkgNameID           bool
+		queryPkgVersionID        bool
+		queryEqualPkgID          bool
+		querySrcTypeID           bool
+		querySrcNamespaceID      bool
+		querySrcNameID           bool
+		queryVulnTypeID          bool
+		queryVulnID              bool
+		queryEqualVulnID         bool
+		queryDeclaredLicenseID   bool
+		queryDiscoveredLicenseID bool
+		certifyBadCall           *certifyBadCall
+		certifyGoodCall          *certifyGoodCall
+		certifyLegalCall         *certifyLegalCall
+		scorecardCall            *scorecardCall
+		vexCall                  *vexCall
+		certifyVulnCall          *certifyVulnCall
+		hashEqualCall            *hashEqualCall
+		hasMetadataCall          *hasMetadataCall
+		hasSBOMCall              *hasSBOMCall
+		hasSlsaCall              *hasSlsaCall
+		hasSourceAtCall          *hasSourceAtCall
+		isDepCall                *isDepCall
+		isOcurCall               *isOcurCall
+		pkgEqualCall             *pkgEqualCall
+		pointOfContactCall       *pointOfContactCall
+		vulnEqualCall            *vulnEqualCall
+		vulnMetadataCall         *vulnMetadataCall
+		usingOnly                []model.Edge
+		want                     []model.Node
+		wantErr                  bool
+	}{{
+		name:           "package - pkgType",
+		pkgInput:       testdata.P1,
+		queryPkgTypeID: true,
+		want: []model.Node{&model.Package{
+			Type: "pypi",
+			Namespaces: []*model.PackageNamespace{{
+				Names: []*model.PackageName{},
+			}}}},
+		wantErr: false,
+	}, {
+		name:                "package - pkgNamespace",
+		pkgInput:            testdata.P1,
+		queryPkgNamespaceID: true,
+		want: []model.Node{
+			&model.Package{
+				Type: "pypi",
+				Namespaces: []*model.PackageNamespace{{
+					Names: []*model.PackageName{{
+						Name:     "tensorflow",
+						Versions: []*model.PackageVersion{},
+					}},
+				}}},
+			&model.Package{
+				Type:       "pypi",
+				Namespaces: []*model.PackageNamespace{},
+			}},
+		wantErr: false,
+	}, {
+		name:           "package - pkgName",
+		pkgInput:       testdata.P1,
+		queryPkgNameID: true,
+		want: []model.Node{testdata.P1out, &model.Package{
+			Type: "pypi",
+			Namespaces: []*model.PackageNamespace{{
+				Names: []*model.PackageName{},
+			}}}},
```

Comment:
Shouldn't this one connect to the pkgversion too?

Reply:
I see `testdata.P1out` now. Small nit is to put each element in the list on its own line then.

Reply:
oops looks like I missed that one. Done!